influxql/iterator.gen.go.tmpl

```diff
@@ -167,6 +164,18 @@ func (itr *{{$k.name}}MergeIterator) Close() error {
 
 // Next returns the next point from the iterator.
 func (itr *{{$k.name}}MergeIterator) Next() *{{$k.Name}}Point {
+	if !itr.init {
+		items := itr.heap.items
+		itr.heap.items = make([]*{{$k.name}}MergeHeapItem, 0, len(items))
+		for _, item := range items {
+			if item.itr.peek() == nil {
+				continue
+			}
+			itr.heap.items = append(itr.heap.items, item)
+		}
+		heap.Init(itr.heap)
+	}
+
```

Comment:
Can you add a comment to the code? I'm sure I won't remember that in six months. :)